numpy/ma/tests/test_core.py

```diff
@@ -4732,6 +4732,56 @@ def test_ctor(self):
         assert_(not isinstance(m, np.ma.core.MaskedConstant))
         assert_(m is not np.ma.masked)
 
+    def test_repr(self):
+        # copies should not exist, but if they do, it should be obvious that
+        # something is wrong
+        assert_equal(repr(np.ma.masked), 'masked')
+        assert_not_equal(repr(np.ma.masked.copy()), 'masked')
+
+    def test_pickle(self):
+        from io import BytesIO
+        import pickle
+
+        with BytesIO() as f:
+            pickle.dump(np.ma.masked, f)
+            f.seek(0)
+            res = pickle.load(f)
+        assert_(res is np.ma.masked)
+
+    def test_write_to_copy(self):
+        # gh-9328
+        x = np.ma.masked.copy()
```

Comment:
Correct, because that seemed the least bad option, with the others being:

* `return self` - this is clearly not a copy, and will fail when trying to write to it, whereas normally you would expect to be able to write to the result of copy() without side-effects
* `return MaskedConstant(self)` - this is super confusing, because it printed as `masked`, and would be neither constant nor singleton